csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -62,11 +62,17 @@ public static void TryDelete(string file)
         /// </summary>
         public static string FindExecutableOnPath(string exe)
         {
-            var isWindows = Win32.IsWindows();
-            var paths = Environment.GetEnvironmentVariable("PATH").Split(isWindows ? ';' : ':');
-            var exes = isWindows
-                ? Environment.GetEnvironmentVariable("PATHEXT").Split(';').Select(ext => exe + ext)
-                : new[] { exe };
+            var paths = Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator);
+            string[] exes;
+            if (Win32.IsWindows())
+            {
+                var extensions = Environment.GetEnvironmentVariable("PATHEXT").Split(';').ToArray();
```

Comment:
Should we add extra handling here in case `GetEnvironmentVariable` returns `null`? It might be good to add this as a source of nullness in the nullness queries.